java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java

```diff
@@ -130,18 +131,34 @@ public void visitClass(ClassTree tree) {
     @Override
     public void visitMethod(MethodTree tree) {
       TypeTree returnTypeTree = tree.returnType();
-      if (isNotOverriding(tree) || returnTypeTree == null) {
+      populateOverridingMethodExclusions(tree, returnTypeTree);
+      if (!isOverriding(tree) || returnTypeTree == null) {
         if (returnTypeTree != null) {
           reportIssueOnDeprecatedType(returnTypeTree, returnTypeTree.symbolType());
         }
         scan(tree.parameters());
       }
       scan(tree.block());
+      overridingMethodTypes.pop();
     }
 
-    private boolean isNotOverriding(MethodTree tree) {
+    void populateOverridingMethodExclusions(MethodTree methodTree, TypeTree returnType) {
+      var methodTypes = new HashSet<String>();
+      if (isOverriding(methodTree)) {
+        if(returnType != null) {
+          methodTypes.add(returnType.symbolType().fullyQualifiedName());          
+        }
+        methodTree.parameters().stream()
+          .filter(param -> param.type() != null)
```

Comment:
Do we know in which case a parameter, which is `VariableTree`, return null on `type()`?